File: listr-svc/ListR/ListR.Common/EntityModels/Lists/ListItem.cs
using ListR.Common.Enums;
using System.ComponentModel.DataAnnotations.Schema;

namespace ListR.DataLayer.EntityModels.Lists
{
    public class ListItem
    {
        public int Id { get; set; }

        public int ListId { get; set; }

        public string Name { get; set; } = string.Empty;

        public ListItemTypeEnum ItemType { get; set; }

        public Priority Priority { get; set; }

        public bool IsCollected { get; set; } = false;

        public decimal Price { get; set; }
    }
}

File: listr_app/ListR/lib/common/models/Data/list_item.dart
class ListItem
{
  int? id;
  int? listId;
  String? name;
  int? itemType;
  int? priority;
  bool? isCollected;
  double? price;

  ListItem(
      {this.id,
      this.listId,
      this.name,
      this.itemType,
      this.priority,
      this.isCollected,
      this.price});

  ListItem.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    listId = json['listId'];
    name = json['name'];
    itemType = json['itemType'];
    priority = json['priority'];
    isCollected = json['isCollected'];
    price = json['price'];
  }

  Map<String, dynamic> toMap() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['listId'] = listId;
    data['name'] = name;
    data['itemType'] = itemType;
    data['priority'] = priority;
    data['isCollected'] = isCollected;
    data['price'] = price;
    return data;
  }
}